File: 0438_Find All Anagrams in a String/FindAllAnagramsinaString.cs
public class Solution {
    public IList<int> FindAnagrams(string s, string p) {
        var ans = new List<int>();
        int m = s.Length, n = p.Length;
        if(m < n) return ans;
        var count = new int[26];
        
        for(int i=0;i<n;i++)
        {
            count[p[i]-'a']++;
            count[s[i]-'a']--;
        }
        
        if(AllZeros(count))
            ans.Add(0);
        
        for(int i=n;i<m;i++)
        {
            count[s[i]-'a']--;
            count[s[i-n]-'a']++;
            if(AllZeros(count))
                ans.Add(i-n+1);
        }
        
        return ans;
    }
    
    private bool AllZeros(int[] count)
    {
        return count.Where(c => c!=0).Count() == 0;
    }
}

File: 0438_Find All Anagrams in a String/FindAllAnagramsinaString.java
class Solution {
    public List<Integer> findAnagrams(String s, String p) {
        List<Integer> ans = new ArrayList<>();
        int[] arr = new int[26];
        for(int i=0;i<p.length();i++) {
            arr[p.charAt(i) - 'a']++;
        }
        
        for(int i=0;i<s.length();i++) {
    
            arr[s.charAt(i) - 'a']--;
            if(i - p.length() >= 0) {
                arr[s.charAt(i-p.length()) - 'a']++;
            }
            
            if(allZeros(arr)) {
                ans.add(i - p.length()+1);
            }
        }
        
        return ans;
    }
    
    private boolean allZeros(int[] arr) {
        for(int i=0;i<arr.length;i++) {
            if(arr[i] != 0) return false;
        }
        
        return true;
    }
}